### PR TITLE
Convmix 

### PR DESCRIPTION
**Pull request by totto82**
This is a rebase of https://github.com/OPM/opm-models/pull/902

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 please

**Comment by totto82** [comment]
benchmark please opm-simulators=5418

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *

FAILED RUNS: 
OPM Benchmark: drogon - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: drogon - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: punqs3 - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: punqs3 - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: smeaheia - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: smeaheia - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access

View result details @ https://www.ytelses.com/opm/?page=result&id=2564

**Comment by totto82** [comment]
benchmark please opm-simulators=5491

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *

FAILED RUNS: 
OPM Benchmark: drogon - Threads: 1 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: drogon - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: drogon - Threads: 8 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: drogon - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: punqs3 - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: punqs3 - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: smeaheia - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: smeaheia - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 1 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 8 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 1 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 8 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 1 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 8 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 - 2024-07-30 Master Reference: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access

View result details @ https://www.ytelses.com/opm/?page=result&id=2566

**Comment by totto82** [comment]
benchmark please opm-simulators=5491

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: drogon - Threads: 1                | 0.999    |
| opm-git | OPM Benchmark: drogon - Threads: 8                | 0.979    |
| opm-git | OPM Benchmark: punqs3 - Threads: 1                | 1.023    |
| opm-git | OPM Benchmark: punqs3 - Threads: 8                | 1.005    |
| opm-git | OPM Benchmark: smeaheia - Threads: 1              | 1.002    |
| opm-git | OPM Benchmark: smeaheia - Threads: 8              | 1.017    |
| opm-git | OPM Benchmark: spe10_model_1 - Threads: 1         | 1.011    |
| opm-git | OPM Benchmark: spe10_model_1 - Threads: 8         | 0.998    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.995    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.003    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 1.016    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.007    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 1.003    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1.008    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=2570

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 please

**Comment by totto82** [comment]
benchmark please opm-simulators=5418

**Comment by totto82** [comment]
benchmark please opm-simulators=5491

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *

FAILED RUNS: 
OPM Benchmark: drogon - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: drogon - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: punqs3 - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: punqs3 - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: smeaheia - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: smeaheia - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access

View result details @ https://www.ytelses.com/opm/?page=result&id=2579

**Comment by totto82** [comment]
It is hard to keep-up with the merges in opm. Benchmark only works if all branches are updated. I will do a last attempt.

**Comment by totto82** [comment]
benchmark please opm-simulators=5491

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *

FAILED RUNS: 
OPM Benchmark: drogon - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: drogon - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: punqs3 - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: punqs3 - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: smeaheia - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: smeaheia - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: spe10_model_1 - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_extra - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access
OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 - opm-models 910: TESTING: Convmix: The test run did not produce a result. E: mpirun was unable to launch the specified application as it could not access

View result details @ https://www.ytelses.com/opm/?page=result&id=2580

File: opm/models/blackoil/blackoilconvectivemixingmodule.hh
Comment (trinemykk) [inline opm/models/blackoil/blackoilconvectivemixingmodule.hh:1]: moving the density calculation here was a good choice, the other changes looks consistent

**Review by trinemykk — APPROVED** [review]
this looks consistent and a good choice to collect the density calculation

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 update_data please

**Comment by totto82** [comment]
jenkins build this opm-simulators=5491 opm-tests=1204 please

**Comment by totto82** [comment]
Finally. All tests pass. Lets get this in.

**Pull request merged**